service/http.directory/src/main/java/org/kaazing/gateway/service/http/directory/HttpDirectoryServiceHandler.java

```diff
@@ -104,6 +104,10 @@ void setPatterns(List<PatternCacheControl> patterns) {
         this.patterns = patterns;
     }
 
+    void emptyUrlCacheControlMap() {
+        urlCacheControlMap.clear();
+    }
```

Comment:
I don't understand. Can you elaborate bit here. There is only one HttpDirectoryServiceHandler instance per service. Then I don't see any need for static unless you want to cache across multiple services. All the requests for a service are handled at one handler. So static map is not required.  What am I missing ?

Also, you haven't commented on thread-safety issue.